controllers/secretscope_controller_databricks.go

```diff
@@ -150,25 +152,74 @@ func (r *SecretScopeReconciler) submitACLs(instance *databricksv1alpha1.SecretSc
 	return nil
 }
 
-func (r *SecretScopeReconciler) submit(instance *databricksv1alpha1.SecretScope) error {
+// verifyWorkspace checks if Databricks cluster supports ACLs, and checks if secret scope exists.
+func (r *SecretScopeReconciler) verifyWorkspace(instance *databricksv1alpha1.SecretScope) error {
 	scope := instance.ObjectMeta.Name
 	initialManagePrincipal := instance.Spec.InitialManagePrincipal
 
-	err := r.APIClient.Secrets().CreateSecretScope(scope, initialManagePrincipal)
+	// try create secret scope to see if exists or not.
+	list, err := r.APIClient.Secrets().ListSecretScopes()
 	if err != nil {
 		return err
 	}
 
-	err = r.submitSecrets(instance)
+	for _, n := range list {
+		if n.Name == scope {
+			return fmt.Errorf("Secret scope %v exists", scope)
+		}
+	}
+
+	// create the secret scope here otherwise we cannot verify ACL support.
+	err = r.APIClient.Secrets().CreateSecretScope(scope, initialManagePrincipal)
 	if err != nil {
 		return err
 	}
 
-	err = r.submitACLs(instance)
+	// try to list ACLs to see if cluster supports ACL.
+	if instance.Spec.SecretScopeACLs != nil {
```

Comment:
I was wondering if we can get Databricks SKU by using azure-sdk-for-go when we install operator  - https://github.com/Azure/azure-sdk-for-go/blob/7a9d2769e4a581b0b1bc609c71b59af043e05c98/services/databricks/mgmt/2018-04-01/databricks/models.go